sys/include/net/gnrc/pkt.h

```diff
@@ -185,8 +195,10 @@ static inline gnrc_pktsnip_t *gnrc_pkt_append(gnrc_pktsnip_t *pkt,
 static inline gnrc_pktsnip_t *gnrc_pkt_prepend(gnrc_pktsnip_t *pkt,
                                                gnrc_pktsnip_t *snip)
 {
-    LL_PREPEND(pkt, snip);
-    return pkt;
+    list_node_t list = { .next = (list_node_t *)pkt };
+
+    list_add(&list, (list_node_t *)snip);
+    return (gnrc_pktsnip_t *)list.next;
```

Comment:
That's a bit too clever for my liking.
Doesn't that just boil down to

```suggestion
    snip->next = pkt;
    return snip;
```

Reply:
Doesn't make a huge difference on code size, as the compiler is smart enough to resolve that.

Reply:
The compiler may be smart enough, but the cursory reader may not.

Reply:
It is not about the compiler being smart enough to optimize here it is about being smart enough that in case `list_add()` is used more often in a unit, it makes it a static function.

Reply:
I agree with @benpicco here. Here `list_node_t list` is allocated on the stack, so that memory become invalid after the function scope is left. I needed quite some time to confirm that `list` is indeed not used anymore once the scope of the function is left.

I'm personally a bit puzzled that this is resulted in smaller ROM size. I would personally prefer to trade in a few bytes of ROM for the increased readability.

Reply:
Ok, done.